manifests/certonly.pp

```diff
@@ -31,20 +31,30 @@
 # [*cron_success_command*]
 #   String representation of a command that should be run if the renewal command
 #   succeeds.
+# [*cron_hour*]
+#   Optional string, integer or array, hour(s) that the renewal command should execute.
+#   e.g. '[0,12]' execute at midnight and midday.  Default - seeded random hour.
+# [*cron_minute*]
+#   Optional string, integer or array, minute(s) that the renewal command should execute.
+#   e.g. 0 or '00' or [0,30].  Default - seeded random minute.
 #
 define letsencrypt::certonly (
-  Array $domains                                            = [$title],
-  Boolean $custom_plugin                                    = false,
-  Enum['apache', 'standalone', 'webroot', 'nginx'] $plugin  = 'standalone',
-  Optional[Array] $webroot_paths                            = undef,
-  String $letsencrypt_command                               = $letsencrypt::command,
-  Optional[Array] $additional_args                          = undef,
-  Array $environment                                        = [],
-  Boolean $manage_cron                                      = false,
-  Boolean $suppress_cron_output                             = false,
-  $cron_before_command                                      = undef,
-  $cron_success_command                                     = undef,
-  Stdlib::Unixpath $config_dir                              = $letsencrypt::config_dir,
+  Array $domains                                           = [$title],
+  Boolean $custom_plugin                                   = false,
+  Enum['apache', 'standalone', 'webroot', 'nginx'] $plugin = 'standalone',
+  Optional[Array] $webroot_paths                           = undef,
+  String $letsencrypt_command                              = $letsencrypt::command,
+  Optional[Array] $additional_args                         = undef,
+  Array $environment                                       = [],
+  Boolean $manage_cron                                     = false,
+  Boolean $suppress_cron_output                            = false,
+  $cron_before_command                                     = undef,
+  $cron_success_command                                    = undef,
+  # 0 - 23, seed is title plus fqdn
+  Variant[Integer, String, Array] $cron_hour               = fqdn_rand(24, $title),
```

Comment:
You can limit it further by specifying `Integer[0, 23]`. At least, I think that was inclusive. Also wondering if the `Array` should be further specified. A similar comment applies to the minute parameter.